src/osd/PrimaryLogPG.cc

```diff
@@ -13967,6 +13984,22 @@ bool PrimaryLogPG::agent_choose_mode(bool restart, OpRequestRef op)
     goto skip_calc;
   }
 
+  if (pool.info.cache_mode == pg_pool_t::CACHEMODE_WRITEBACK &&
+      pool.info.cache_pin_head) {
+    // do not flush omap objects if ec backing pool
+    const pg_pool_t *base_pool = get_osdmap()->get_pg_pool(pool.info.tier_of);
+    assert(base_pool);
+    uint64_t unflushable = base_pool->supports_omap() ?
+      0 : info.stats.stats.sum.num_objects_omap;
+
+    if (info.stats.stats.sum.num_whiteouts > 0 ||
+        info.stats.stats.sum.num_object_clones > unflushable) {
```

Comment:
why compare num_object_clones with num_objects_omap, why not num_object_clones > 0, what is the relationship with them

Reply:
the ec pool is not supported for omap